pkg/labels/selector.go

```diff
@@ -600,7 +600,12 @@ func (p *Parser) parseIdentifiersList() (sets.String, error) {
 // parseExactValue parses the only value for exact match style
 func (p *Parser) parseExactValue() (sets.String, error) {
 	s := sets.NewString()
-	tok, lit := p.consume(Values)
+	tok, lit := p.lookahead(Values)
+	if tok == EndOfStringToken {
+		s.Insert(lit)
+		return s, nil
+	}
+	tok, lit = p.consume(Values)
```

Comment:
No. `consume()` will increments the position in the token array. Later this will cause an `index out of range` panic because the EndoOfStringToken is consumed.